### PR TITLE
gameontree

### PR DESCRIPTION
**Pull request by Keys9873**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [x] I have linked this PR to any issues that it closes.

### CODE REVIEW

File: solutions/advanced/ac-GameOnTree.mdx [hunk 1]
```diff
@@ -0,0 +1,57 @@
+---
+id: ac-GameOnTree
+source: AtCoder Grand Contest
+title: Game On Tree
+author: Aakash Gokhale
+---
+
+[Official Editorial (Japanese)](https://atcoder.jp/contests/agc017/editorial)
+
+Since the game in the problem seems similar to nim, let's try to use sprague grundy numbers.
```
Comment (SansPapyrus683) [inline solutions/advanced/ac-GameOnTree.mdx:10]: pretty sure these should be proper nouns

File: solutions/advanced/ac-GameOnTree.mdx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+To solve the problem, for a certain root with subtees, $s_1, s_2, ..., s_k$, we can try to convert all these subtrees into a stack of some size.
```
Comment (SansPapyrus683) [inline solutions/advanced/ac-GameOnTree.mdx:12]: > convert all these subtrees into a stack of some size.

you can't just gloss past this 😭

File: solutions/advanced/ac-GameOnTree.mdx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+int main() {
+	ios::sync_with_stdio(0);
+	cin.tie(0);
```
Comment (SansPapyrus683) [inline solutions/advanced/ac-GameOnTree.mdx:32]: remove if runs w/o

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/advanced/ac-GameOnTree.mdx (hunk 2 above)
Comment (Keys9873) [reply solutions/advanced/ac-GameOnTree.mdx:12]: is it better now?

Comment (SansPapyrus683) [reply solutions/advanced/ac-GameOnTree.mdx:12]: > subtees

File: solutions/advanced/ac-GameOnTree.mdx [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+If the grundy value of the root is positive, then Alice wins, otherwise if the grundy value of the root is $0$, Bob wins.
+
+Notice that when we get the grundy numbers for each subtree we need to add $1$ to all of them before xoring.
+This is because each stack's size increases by $1$ when you consider the edge from the root of $s_i$ to the root of the entire tree.
+
+For example consider the tree:
+
+$3 \\ $
+$1$ $2 \\ $
+$1$ $3 \\ $
```
Comment (SansPapyrus683) [inline solutions/advanced/ac-GameOnTree.mdx:27]: come on just make a diagram at this point

File: solutions/advanced/ac-GameOnTree.mdx [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+	int n;
+	cin >> n;
+	vector<vector<int>> g(n + 1);
+	for (int i = 0; i < n - 1; i++) {
+		int u, v;
+		cin >> u >> v;
+		g[u].push_back(v);
+		g[v].push_back(u);
+	}
+	auto dfs = [&](auto self, int u, int p = 0) -> int {
```
Comment (SansPapyrus683) [inline solutions/advanced/ac-GameOnTree.mdx:57]: just use a type declaration come on `self` makes it so much more unreadable

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

**Review by ryanchou-dev — APPROVED** [review]
🙉

**Comment by TheGamingMousse** [comment]
maybe make it 0 indexed? does not really matter tho ig

File: solutions/advanced/ac-GameOnTree.mdx (hunk 4 above)
Comment (SansPapyrus683) [reply solutions/advanced/ac-GameOnTree.mdx:27]: bump

File: solutions/advanced/ac-GameOnTree.mdx [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+```cpp
+#include <iostream>
+#include <vector>
+
+using namespace std;
+
+const int N = 1e5 + 5;
+vector<vector<int>> g(N);
+
+int dfs(int u, int p = 0) {
```
Comment (SansPapyrus683) [inline solutions/advanced/ac-GameOnTree.mdx:49]: not lambda yet

File: solutions/advanced/ac-GameOnTree.mdx (hunk 4 above)
Comment (ryanchou-dev) [reply solutions/advanced/ac-GameOnTree.mdx:27]: 😦

how's this?
image

Comment (Keys9873) [reply solutions/advanced/ac-GameOnTree.mdx:27]: fire

File: solutions/advanced/ac-GameOnTree.mdx (hunk 6 above)
Comment (Keys9873) [reply solutions/advanced/ac-GameOnTree.mdx:49]: i think std::function is ugly
is this not fine

Comment (SansPapyrus683) [reply solutions/advanced/ac-GameOnTree.mdx:47]: if justin can do it so can you

Comment (Keys9873) [reply solutions/advanced/ac-GameOnTree.mdx]: if u insist i guess

**Review by SansPapyrus683 — APPROVED** [review]

**Pull request merged**